modin/experimental/xgboost/xgboost_ray.py

```diff
@@ -680,37 +604,39 @@ def _predict(
     """
     s = time.time()
 
-    X_row_parts, _ = data
-
-    num_actors = _get_num_actors(
-        num_actors if isinstance(num_actors, int) else "default_predict"
-    )
-
-    if num_actors > len(X_row_parts):
-        num_actors = len(X_row_parts)
-
-    # Create remote actors
-    actors = create_actors(num_actors)
+    # Get metainfo from dmatrix
+    input_index, input_columns, row_lengths = data.data_metainfo
 
-    # Split data across workers
-    _split_data_across_actors(
-        actors,
-        lambda actor, *X: actor.set_predict_data.remote(*X),
-        X_row_parts,
-        is_predict=True,
-    )
+    # Infer columns of result
+    def _get_num_columns(booster, n_features, **kwargs):
```

Comment:
What logic is here?

Reply:
It's needed for getting the number of columns of prediction result. We do fast prediction on the one-row dataset
```python
def _get_num_columns(booster, n_features, **kwargs):
        rng = np.random.RandomState(777)
        test_data = rng.randn(1, n_features) # one-row dataset
        test_predictions = booster.predict(
            xgb.DMatrix(test_data), validate_features=False, **kwargs
        ) # fast predict
        num_columns = (
            test_predictions.shape[1] if len(test_predictions.shape) > 1 else 1
        ) # getting of columns number
        return num_columns
```

Reply:
If we put this in a remote call, we should add `ray.get` for returned `ObjectRef`, because materialized `num_columns` is used further for creating `new_columns`. Time increases in this case from 0.19 s to 0.28 s (average between 10 runs for higgs data). Possible reason of slowdown is too small task  `_get_num_columns`, in that case overheads to submitting may exceed task duration.

Reply:
makes sense